src/js/components/FileInput/FileInput.js

```diff
@@ -134,129 +134,288 @@ const FileInput = forwardRef(
     }
 
     return (
-      <Keyboard
-        onSpace={event => {
-          if (controlRef.current === event.target) inputRef.current.click();
-        }}
-        onEnter={event => {
-          if (controlRef.current === event.target) inputRef.current.click();
-        }}
-      >
-        <ContentsBox
-          ref={controlRef}
-          tabIndex={!plain ? 0 : undefined}
-          theme={theme}
-          disabled={disabled}
-          background={mergeTheme('background', 'color')}
-          border={mergeTheme('border', 'side')}
-          margin={mergeTheme('margin')}
-          pad={mergeTheme('pad')}
-          round={mergeTheme('round', 'size')}
-          align={files.length ? 'stretch' : 'center'}
-          justify="center"
-          hover={hover}
-          onMouseOver={() => setHover(true)}
-          onMouseOut={() => setHover(false)}
-          dragOver={dragOver}
-        >
-          {files.length > aggregateThreshold && (
-            <Box direction="row" align="center" justify="between">
-              <Label {...theme.fileInput.label}>
-                {files.length} {messages.files || 'files'}
-              </Label>
-              <Button
-                ref={removeRef}
-                a11yTitle={messages.removeAll || 'remove all'}
-                icon={<RemoveIcon />}
-                hoverIndicator
-                onClick={event => {
-                  event.stopPropagation();
-                  setFiles([]);
-                  inputRef.current.focus();
-                }}
-              />
-            </Box>
-          )}
-          {files.length > 0 &&
-            files.length <= aggregateThreshold &&
-            files.map((file, index) => (
-              <Box
-                key={file.name}
-                direction="row"
-                align="center"
-                justify="between"
-              >
-                {renderFile ? (
-                  renderFile(file)
-                ) : (
-                  <Label
-                    weight={
-                      theme.global.input.weight ||
-                      theme.global.input.font.weight
-                    }
-                    truncate
-                    {...theme.fileInput.label}
-                  >
-                    {file.name}
-                  </Label>
-                )}
+      <>
+        {theme.fileInput.button ? (
```

Comment:
Can we do this with less code duplication? For instance, the ContentsBox is duplicated in both parts of the conditional.